src/autowiring/test/OnceTest.cpp

```diff
@@ -55,6 +55,8 @@ TEST(OnceTest, MultiLambdaPending) {
     while(go) {
       o += [&] { x++; };
       y++;
+      if (y % 10000 == 0)
+        std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
When I change this to yield() it still hangs, likewise for a similar workaround in AutoSignalTest.. Apparently valgrind can choose to ignore yields.

Just now I learned that valgrind has an option `--fair-sched=yes` which better mimics reality and can get these tests to complete without sleep_for(). Given the nature of some these and some other autowiring unit tests, perhaps this should be mandatory for running AutowiringTest in valgrind.